jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -360,8 +393,10 @@ protected HiveConnection(String uri, Properties info,
         throw new SQLException(new IllegalArgumentException(
             "Browser mode is not supported in embedded mode"));
       }
-      openSession();
-      executeInitSql();
+      if (initSession) {
```

Comment:
could we extract this into a helper method?

Reply:
I don't think it makes sense to add a function here.